libmachine/provision/ubuntu_systemd.go

```diff
@@ -0,0 +1,233 @@
+package provision
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"text/template"
+
+	"github.com/docker/machine/libmachine/auth"
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/engine"
+	"github.com/docker/machine/libmachine/log"
+	"github.com/docker/machine/libmachine/mcnutils"
+	"github.com/docker/machine/libmachine/provision/pkgaction"
+	"github.com/docker/machine/libmachine/provision/serviceaction"
+	"github.com/docker/machine/libmachine/swarm"
+)
+
+func init() {
+	Register("Ubuntu-SystemD", &RegisteredProvisioner{
+		New: NewUbuntuSystemDProvisioner,
+	})
+}
+
+func NewUbuntuSystemDProvisioner(d drivers.Driver) Provisioner {
+	return &UbuntuSystemDProvisioner{
+		GenericProvisioner{
+			DockerOptionsDir:  "/etc/docker",
+			DaemonOptionsFile: "/etc/systemd/system/docker.service",
+			OsReleaseID:       "ubuntu",
+			Packages: []string{
+				"curl",
+			},
+			Driver: d,
+		},
+	}
+}
+
+type UbuntuSystemDProvisioner struct {
+	GenericProvisioner
+}
+
+func (provisioner *UbuntuSystemDProvisioner) CompatibleWithHost() bool {
+	isUbuntu := provisioner.OsReleaseInfo.ID == provisioner.OsReleaseID
+	if !isUbuntu {
+		return false
+	}
+	versionNumber, err := strconv.ParseFloat(provisioner.OsReleaseInfo.VersionID, 32)
+	if err != nil {
+		return false
+	}
+	return versionNumber >= 15
```

Comment:
Should specifically be `15.04`, right?  Can you please specify that as a `const` and compare against that? E.g.:

```
func () ... {
const FirstUbuntuSystemdVersion = 15.04
....
return versionNumber >= FirstUbuntuSystemdVersion
```